services/provider/server/server.go

```diff
@@ -717,22 +731,71 @@ func (s *OCSProviderServer) GetStorageClaimConfig(ctx context.Context, req *pb.S
-					Data: mustMarshal(rbdStorageClassData),
+					Name:   "ceph-rbd",
+					Kind:   "StorageClass",
+					Data:   mustMarshal(rbdStorageClassData),
+					Labels: getExternalResourceLabels("StorageClass", mirroringEnabled, false, replicationID, storageID),
 				},
 				&pb.ExternalResource{
 					Name: "ceph-rbd",
 					Kind: "VolumeSnapshotClass",
 					Data: mustMarshal(map[string]string{
 						"csi.storage.k8s.io/snapshotter-secret-name": provisionerSecretName,
-					})},
+					}),
+					Labels: getExternalResourceLabels("VolumeSnapshotClass", mirroringEnabled, false, replicationID, storageID)},
 				&pb.ExternalResource{
 					Name: "ceph-rbd",
 					Kind: "VolumeGroupSnapshotClass",
 					Data: mustMarshal(map[string]string{
 						"csi.storage.k8s.io/group-snapshotter-secret-name": provisionerSecretName,
-					})},
+					}),
+					Labels: getExternalResourceLabels("VolumeGroupSnapshotClass", mirroringEnabled, false, replicationID, storageID)},
+				&pb.ExternalResource{
+					Name: "ceph-rbd",
+					Kind: "VolumeReplicationClass",
+					Data: mustMarshal(map[string]string{
+						"replication.storage.openshift.io/replication-secret-name": provisionerSecretName,
+						"mirroringMode": "snapshot",
+					}),
+					Labels: getExternalResourceLabels("VolumeReplicationClass", mirroringEnabled, false, replicationID, storageID),
+					Annotations: map[string]string{
+						"replication.storage.openshift.io/is-default-class": "true",
+					},
+				},
+				&pb.ExternalResource{
+					Name: "ceph-rbd-flatten",
+					Kind: "VolumeReplicationClass",
+					Data: mustMarshal(map[string]string{
+						"replication.storage.openshift.io/replication-secret-name": provisionerSecretName,
+						"mirroringMode": "snapshot",
+						"flattenMode":   "force",
+					}),
+					Labels:      getExternalResourceLabels("VolumeReplicationClass", mirroringEnabled, true, replicationID, storageID),
+					Annotations: map[string]string{},
+				},
+				&pb.ExternalResource{
+					Name: "ceph-rbd",
+					Kind: "VolumeGroupReplicationClass",
+					Data: mustMarshal(map[string]string{
+						"replication.storage.openshift.io/group-replication-secret-name": provisionerSecretName,
+						"mirroringMode": "snapshot",
+					}),
+					Labels: getExternalResourceLabels("VolumeGroupReplicationClass", mirroringEnabled, false, replicationID, storageID),
+					Annotations: map[string]string{
+						"replication.storage.openshift.io/is-default-class": "true",
+					},
+				},
+				&pb.ExternalResource{
+					Name: "ceph-rbd-flatten",
+					Kind: "VolumeGroupReplicationClass",
+					Data: mustMarshal(map[string]string{
+						"replication.storage.openshift.io/group-replication-secret-name": provisionerSecretName,
+						"mirroringMode": "snapshot",
+						"flattenMode":   "force",
+					}),
+					Labels:      getExternalResourceLabels("VolumeGroupReplicationClass", mirroringEnabled, true, replicationID, storageID),
+					Annotations: map[string]string{},
+				},
```

Comment:
we dont have voluemegroupreplicationclass in 4.18 isnt it? can we remove this and add it later?

Reply:
@rewantsoni